src/EditorFeatures/CSharpTest/ConvertAutoPropertyToFullProperty/ConvertAutoPropertyToFullPropertyTests.cs

```diff
@@ -858,6 +858,98 @@ class TestClass
             await TestMissingAsync(text);
         }
 
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task CursorInType()
+        {
+            var text = @"
+class TestClass
+{
+    public in[||]t Goo { get; set; }
+}
+";
+            var expected = @"
+class TestClass
+{
+    private int goo;
+
+    public int Goo
+    {
+        get
+        {
+            return goo;
+        }
+        set
+        {
+            goo = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [WorkItem(35180, "https://github.com/dotnet/roslyn/issues/35180")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task SelectionWhole()
+        {
+            var text = @"
+class TestClass
+{
+    [|public int Goo { get; set; }|]
+}
+";
+            var expected = @"
+class TestClass
+{
+    private int goo;
+
+    public int Goo
+    {
+        get
+        {
+            return goo;
+        }
+        set
+        {
+            goo = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
+        public async Task SelectionName()
+        {
+            var text = @"
+class TestClass
+{
+    public int [|Goo|] { get; set; }
+}
+";
+            var expected = @"
+class TestClass
+{
+    private int goo;
+
+    public int Goo
+    {
+        get
+        {
+            return goo;
+        }
+        set
+        {
+            goo = value;
+        }
+    }
+}
+";
+            await TestInRegularAndScriptAsync(text, expected, options: DoNotPreferExpressionBodiedAccessors);
+        }
+
         [Fact, Trait(Traits.Feature, Traits.Features.ConvertAutoPropertyToFullProperty)]
```

Comment:
no new tests for selecting attributes?

Reply:
Attributes are thoroughly tested in PullUpMember. I'm not sure how worth is it to test all permutations everywhere all the time.

Reply:
gotcha.  plz see comment about we could likely *reduce* our selection testing investment at the feature levle, and consolidate in a specific test harness for this lower level layer.